doc/dev/testing/azure.md

```diff
@@ -0,0 +1,35 @@
+This document explains how to run tests for etcd-operator functionality when using Azure Blob Storage (ABS) backups
+
+## Unit Tests
```

Comment:
It is unnecessary to add this section.
`go test` will be run on each directory anyway.
It's a matter of what developers should know, e.g. setting up env.
Let's remove this section